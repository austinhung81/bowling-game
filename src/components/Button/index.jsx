import React from "react";
import "./index.scss";

const Button = ({ children, ...rest }) => {
  return (
    <button type="button" className="btn" {...rest}>
      {children}
    </button>
  );
};

export default Button;