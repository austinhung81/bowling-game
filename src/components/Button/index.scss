.btn {
  display: flex;
  text-align: center;
  justify-content: center;
  background-color: #ffffff;
  border: 1px solid rgb(201, 201, 201);
  color: #202124;
  font-size: 14px;
  border-radius: 5px;
  padding: 5px;
  line-height: 20px;
  &:hover {
    cursor: pointer;
    background-color: #f5f5f5;
  }
  &:focus {
    border: 1px solid #1e5ede;
  }
  &:disabled,
  &[disabled] {
    opacity: 0.5;
    cursor: not-allowed;
    background-color: #ffffff;
    &:hover {
      background-color: #ffffff;
    }
  }
  .icon {
    margin-right: 8px;
    margin-top: 2px;
  }
}