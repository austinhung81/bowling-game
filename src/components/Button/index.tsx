import React from "react";
import "./index.scss";

type Props = {
  [key: string]: any;
}

const Button = ({ children, ...rest }: Props): JSX.Element => {
  return (
    <button type="button" className="btn" {...rest}>
      {children}
    </button>
  );
};

export default Button;