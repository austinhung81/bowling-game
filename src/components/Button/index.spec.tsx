
import React from 'react';
import { create } from "react-test-renderer";
import Button from './index';

describe('test Button component', () => {
 it('should be defined', () => {
   expect(Button).toBeDefined();
 });
 it('should match snapshot', () => {
  expect.assertions(1);
  const button = create(<Button className="test" />).toJSON();
  expect(button).toMatchSnapshot();
 });
});