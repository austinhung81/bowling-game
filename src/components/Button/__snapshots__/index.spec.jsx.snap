// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button component should match snapshot 1`] = `
<button
  className="test"
  type="button"
/>
`;
